Extends the clothing booth catalogue

Carbadox opened the pull request:
## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
Adds more items to the clothing booth, namely

- Accessories
Monocole
French scarf
Suspenders

- Casual
Dirty Vest (Shitty Bills vest)
Band Shirt
Blue / Orange hoodies

- Costumes
Russian / Columbian Mobster Suits
Mobile Suit / Mobile Suit Helmet

- Formal
Shirt with black/brown/white pants variants with red/black/blue ties variants respectively
Waistcoat
Butler's Suit
Maid's Outfit

- Jackets
Johnny's Jacket (Overcoat with scarf)
Merchant's Jacket

- Headwear
Top hat
Black / brown / white fedora
Bowler hat
Beret
Cowboy hat

## Why's this needed? <!-- Describe why you think this should be added to the game. -->
Feature enhancement and because playing dress up is fun, regardless of gender and age...right?

## Changelog
```
(u)Carbadox
(+)Added new items to the clothing booth, namely to the accessories, casual, costumes, formal, jackets and the new headwear section.
(+)Added Clothing Booth to manta
```

Carbadox commented:
Feedback would be appreciated if that's alright @AwkwardDryad

Adharainspace commented on code/obj/item/clothing/uniforms.dm:
```diff
@@ -897,18 +897,86 @@
-	desc = "A button-down shirt and some pants."
+//Black Pants
+/obj/item/clothing/under/shirt_pants_b
+	name = "shirt and black pants"
+	desc = "A button-down shirt and some black pants."
 	icon = 'icons/obj/clothing/uniforms/item_js_misc.dmi'
 	wear_image_icon = 'icons/mob/jumpsuits/worn_js_misc.dmi'
 	icon_state = "shirt_n_pant-b"
 	item_state = "shirt_n_pant-b"
 
-	New()
-		..()
-		src.icon_state = "shirt_n_pant-[pick("b", "br", "w")][pick("", "_tie-r", "_tie-b", "_tie-bl")]"
-		src.item_state = "[src.icon_state]"
```
i think making a separate outfit for random generated shirtpants and leaving that as the default for jobs that had it, and leaving all these options for the vendor. gives the jobs a bit of (small) personality

Carbadox replied:
Good point

Carbadox replied:
Done

Adharainspace commented:
map diffs look fine, destiny and manta xmas are formatted weird but it doesnt seem to cause any issues?

Carbadox commented on code/obj/item/clothing/hats.dm:
```diff
@@ -1168,7 +1168,7 @@
 
 /obj/item/clothing/head/pokervisor
-	name = "green visor"
+	name = "Green Visor"
```
It's to follow the same formatting as the other items in the booth but I just realised I pointlessly capitalized all the first letters in the items code and not in its booth entry. Woops

Carbadox replied:
Done

Carbadox commented on code/obj/item/clothing/uniforms.dm:
```diff
@@ -897,18 +897,86 @@
-/obj/item/clothing/under/shirt_pants
-	name = "shirt and pants"
-	desc = "A button-down shirt and some pants."
+//Black Pants
```
//Brown Pants and //White Pants show up later down the line as well, they're there to make it easier to differentiate between all the variants. I think it might be worth keeping in for easier future reference though I'm clueless in code ettiquite so correct me if I'm wrong.

mordent-goonstation replied:
"Section" comments are nice for keeping logical blocks of code together. This entire file (near enough) is basically just a big list of individual, technically unrelated items (with some groupings here and there). Basically, we don't have any official code style or anything that says one way or the other. I would at least make the comments the same way they're done elsewhere in the file (e.g. `// Brown Pants`, with a space after the `//`), so the file is consistent.

comradef191 commented:
![image](https://user-images.githubusercontent.com/62987296/88704994-d1515a00-d106-11ea-9638-0b55424c9370.png)
Manta actually already has a Clothing Booth!

Carbadox commented:
> 
> 
> ![image](https://user-images.githubusercontent.com/62987296/88704994-d1515a00-d106-11ea-9638-0b55424c9370.png)
> Manta actually already has a Clothing Booth!

Hmm, I didn't notice it at first but I do sort of prefer my placement because the one in the image posted is 1. not in public display so not everyone might know there is a clothing booth and 2. it's locked behind doors that aren't all access so the placement is sort of shooting itself in the foot.

For reference, the place where I put it is in the between the 2 wooden wardrobes in the clothing store

comradef191 commented:
> > (Image Snipped)
> > Manta actually already has a Clothing Booth!
> 
> Hmm, I didn't notice it at first but I do sort of prefer my placement because the one in the image posted is 1. not in public display so not everyone might know there is a clothing booth and 2. it's locked behind doors that aren't all access so the placement is sort of shooting itself in the foot.
> 
> For reference, the place where I put it is in the between the 2 wooden wardrobes in the clothing store

Uhh Actually those doors have no access markers meaning everyone can open them, Plus you'd need to make another change that removes the old booth and update the changelog if you want to use your current location

AwkwardDryad commented:
> Feedback would be appreciated if that's alright @AwkwardDryad

the formatting of the booth items looks great as long as it renders nicely in the ui :heart:

Carbadox commented:
okay so summary of changes, corrected green visor letter formatting and properly added it as a booth entry, reverted the map changes and put the random pants/tie variation thing back and made it so that it works with the new stuff and moved the manta clothing booth (no longer in the old position now).

everything is now a-okay for merging, unless someone has any objections

mordent-goonstation commented on assets/maps/prefabs/prefab_water_blindpig.dmm:
```diff
@@ -1,107 +1,107 @@
-ababababababababaaaaaaaaaaaaaaabababababababab
-ababababababababababababababababababababababab
-"}
+"aa" = (/turf/simulated/wall/auto/reinforced/supernorn,/area/blind_pig)
```
Unsure why this prefab changed (could be a subtle change in here that I'm missing). If unintended, could you undo?

Carbadox replied:
this was already reverted using git checkout. The formatting was wack when I added a change (only adding _b at the end shirts and pants) so

mordent-goonstation commented on code/modules/vending/clothingbooth_items.dm:
```diff
@@ -84,21 +126,106 @@
-	path = /obj/item/clothing/under/misc/dress
+
+/datum/clothingbooth_item/shirtandpantsblack
+	name = "Shirt And Black Pants"
```
Looks like convention is to not capitalize "and" (i.e. "Shirt and Black Plants"). Looking at e.g "Skull Mask and Cloak", "Shirt and Red Tie".

Carbadox replied:
done

Carbadox commented on code/modules/vending/clothingbooth_items.dm:
```diff
@@ -340,53 +510,108 @@
+	category = "Headwear"
+	cost = 75
+
+
```
done

Carbadox commented on code/modules/vending/clothingbooth_items.dm:
```diff
@@ -84,21 +126,106 @@
-	name = "Little Red Dress"
-	path = /obj/item/clothing/under/misc/dress/red
+/datum/clothingbooth_item/shirtandpantsblack_redtie
+	name = "Shirt And Black Pants With Red Tie"
```
done
